registry/registry.go

```diff
@@ -323,3 +324,24 @@ type ReconfigurableService interface {
 	// Reconfigure reconfigures the resource
 	Reconfigure(ctx context.Context, cfg config.Service, deps Dependencies) error
 }
+
+// DependencyTypeError is used when a resource doesn't implement the expected interface.
+func DependencyTypeError[T any](name string, actual interface{}) error {
+	return utils.DependencyTypeError(name, (*T)(nil), actual)
+}
```

Comment:
would move/combine this with the one in utils so we only have one DependencyTypeError